tests/typ/meta/diag.typ

```diff
@@ -0,0 +1,7 @@
+// Test diagnostics.
+// Ref: false
+
+---
+// Error: 1:17-1:19 expected length, found integer
```

Comment:
This newline is the reason the test is failing, the test infrastructure could be changed or the message replaced by the hint?

Reply:
for now i would just write something like "expected length, found integer: a length needs a unit – did you mean 12pt?" long-term we can have a more elaborate diagnostics structure with hints